Server/src/services/resources/editor_state.py

```
@@ -1,51 +1,305 @@
-    data: EditorStateData = EditorStateData()
+
+def _now_unix_ms() -> int:
+    return int(time.time() * 1000)
+
+
+def _in_pytest() -> bool:
+    # Avoid instance-discovery side effects during the Python integration test suite.
+    return bool(os.environ.get("PYTEST_CURRENT_TEST"))
+
+
+async def _infer_single_instance_id(ctx: Context) -> str | None:
```

Ruff (0.14.10)

Unused function argument: `ctx`

(ARG001)